src/Microsoft.Health.Fhir.Liquid.Converter/Processors/BaseProcessor.cs

```diff
@@ -38,13 +37,13 @@ public string Convert(string data, string rootTemplate, ITemplateProvider templa
         protected virtual Context CreateContext(ITemplateProvider templateProvider, IDictionary<string, object> data)
         {
             // Load data and templates
-            var timeout = _settings?.TimeOut ?? 0;
+            var timeout = Settings?.TimeOut ?? 0;
             var context = new Context(
                 environments: new List<Hash> { Hash.FromDictionary(data) },
                 outerScope: new Hash(),
                 registers: Hash.FromDictionary(new Dictionary<string, object> { { "file_system", templateProvider.GetTemplateFileSystem() } }),
                 errorsOutputMode: ErrorsOutputMode.Rethrow,
-                maxIterations: _maxIterations,
+                maxIterations: Settings.MaxIterations,
```

Comment:
Settings?.MaxIterations ?? 100000
(optional but for consistency with Timeout)

Reply:
i think setting should be required here.

Reply:
Updated, made the ```processorSettings``` required in processors constructor